care/facility/api/serializers/consultation_symptom.py

```diff
@@ -0,0 +1,125 @@
+from copy import copy
+
+from django.db import transaction
+from rest_framework import serializers
+
+from care.facility.events.handler import create_consultation_events
+from care.facility.models.consultation_symptom import (
+    ClinicalImpressionStatus,
+    ConsultationSymptom,
+    Symptom,
+)
+from care.users.api.serializers.user import UserBaseMinimumSerializer
+
+
+class ConsultationSymptomSerializer(serializers.ModelSerializer):
+    id = serializers.UUIDField(source="external_id", read_only=True)
+    created_by = UserBaseMinimumSerializer(read_only=True)
+    updated_by = UserBaseMinimumSerializer(read_only=True)
+
+    class Meta:
+        model = ConsultationSymptom
+        exclude = (
+            "consultation",
+            "external_id",
+            "deleted",
+        )
+        read_only_fields = (
+            "created_date",
+            "modified_date",
+            "is_migrated",
+        )
+
+    def validate(self, attrs):
+        validated_data = super().validate(attrs)
+        consultation = (
+            self.instance.consultation
+            if self.instance
+            else self.context["consultation"]
+        )
+
+        onset_date = (
+            self.instance.onset_date
+            if self.instance
+            else validated_data.get("onset_date")
+        )
+        if cure_date := validated_data.get("cure_date"):
+            if cure_date < onset_date:
+                raise serializers.ValidationError(
+                    {"cure_date": "Cure date should be after onset date"}
+                )
+
+        if validated_data.get("symptom") != Symptom.OTHERS and validated_data.get(
+            "other_symptom"
+        ):
+            raise serializers.ValidationError(
+                {
+                    "other_symptom": "Other symptom should be empty when symptom is not OTHERS"
+                }
+            )
+
+        if validated_data.get("symptom") == Symptom.OTHERS and not validated_data.get(
+            "other_symptom"
+        ):
+            raise serializers.ValidationError(
+                {
+                    "other_symptom": "Other symptom should not be empty when symptom is OTHERS"
+                }
+            )
+
+        if (
+            ConsultationSymptom.objects.filter(
+                consultation=consultation,
+                symptom=validated_data.get("symptom"),
+                other_symptom=validated_data.get("other_symptom") or "",
+                cure_date__isnull=True,
+            )
+            .exclude(
+                clinical_impression_status=ClinicalImpressionStatus.ENTERED_IN_ERROR
+            )
```

Comment:
We could simply filter out by IN_PROGRESS status right?

```suggestion
            ConsultationSymptom.objects.filter(
                consultation=consultation,
                symptom=validated_data.get("symptom"),
                other_symptom=validated_data.get("other_symptom") or "",
                clinical_impression_status=ClinicalImpressionStatus.IN_PROGRESS,
            )
```